ament_package/package.py

```diff
@@ -98,9 +98,21 @@ def validate(self):
 
         if not self.name:
             errors.append('Package name must not be empty')
-        # only allow lower case alphanummeric characters and underscores
+        # assume packages are catkin unless a build_type is exported.
+        build_type = 'catkin'
+        build_types = [ex.contents for ex in self.exports if ex.tagname == 'build_type']
+        if len(build_types) > 1:
+            # this is bad, but first I must ask if we should make this an error.
```

Comment:
The referenced `catkin_pkg` PR adds the method `get_build_type` for that reason which should be replicated here.